### PR TITLE
Feature to collect deployment metrics from Jenkins

### PR DESCRIPTION
**Pull request by premyadav1**
This feature is an alternate solution to HygieiaJenkinsPlugin. An enterprise might restrict installation of Hygieia Jenkins Plungin for a number of reasons - compliance, stability, governance, SOX etc...
HygieiaJenkinsPluginAlt is called by Jenkins pipeline, it capture metrics (same as collected by Hygieia Jenkins plugin)  and push it to Hygieia by calling Hygieia API's to persist data in Hygieia Mongo database. 

Please refer readme document for details.

### CODE REVIEW

**Comment by shriver135** [comment]
@premyadav1 
Can you separate out the pull request into the code into this branch, and convert the word doc into a markdown file. We place all relevant documentation for users in the gh-pages branch:
https://github.com/Hygieia/Hygieia/tree/gh-pages/pages/hygieia/collectors

**Comment by premyadav1** [comment]
Created pull request to add documentation for HygieiaJenkinsPluginAlt within gh-pages. Here is the pull request url https://github.com/Hygieia/Hygieia/pull/2785

File: hygieia-jenkins-plugin-alt/src/main/hygieia/HygieiaJenkinsPluginAlt.groovy [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+		  {  
+			 "scmRevisionNumber":"$scmRevisionNumber",
+			 "scmCommitLog":"$scmChangedDescription",
+			 "scmAuthor":"$scmCommitDeveloper",
+			 "scmCommitTimestamp":"$scmCommitTimestamp",
+			 "numberOfChanges":"$noOfFilesChanged"
+		  }
+	   ]
+	}"""
+    def response = httpRequest httpMode: 'POST', requestBody: contents, customHeaders: customHeaders, url: hygieiaApiUrl
```
Comment (rvema) [inline hygieia-jenkins-plugin-alt/src/main/hygieia/HygieiaJenkinsPluginAlt.groovy]: Should the hygieiaApiUrl be defined in the Jenkins global properties or at the pipeline step? Also do you need supporting apis for the uri /deploy and /artifact for this client to work?

**Comment by rvema** [comment]
@premyadav1 Any update on the review comments?

File: hygieia-jenkins-plugin-alt/src/main/hygieia/HygieiaJenkinsPluginAlt.groovy (hunk 1 above)
Comment (premyadav1) [reply hygieia-jenkins-plugin-alt/src/main/hygieia/HygieiaJenkinsPluginAlt.groovy]: Build, artifact and deploy api's are available within Hygieia API. HygieiaApiUrl should point to your hygieia api, url will be like - https://hygieiaapihost:port/api/build, https://hygieiaapihost:port/api/artifact and https://hygieiaapihost:port/api/deploy. No additional supporting api's are need. I recommend using pipelineascode - Jenkinsfile and declare the hygieiaApiUrl within that file.

Comment (rvema) [reply hygieia-jenkins-plugin-alt/src/main/hygieia/HygieiaJenkinsPluginAlt.groovy]: ok, however the version of apis invoked seemed to be little dated. However lets leave the discretion to users to choose the endpoints. Can you update the documentation for these as well?

**Pull request merged**